pkg/operatorsource/catalogsourceconfigcrud.go

```diff
@@ -0,0 +1,72 @@
+package operatorsource
+
+import (
+	"context"
+
+	v1 "github.com/operator-framework/operator-marketplace/pkg/apis/operators/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// CatalogSourceConfigCRUD defines CRUD operations for CatalogSourceConfig object in the cluster
+type CatalogSourceConfigCRUD interface {
```

Comment:
I would go a step further and say that this interface can be promoted into a generalized client wrapper so that it can be reused:
```
type Client Interface {
   Update(ctx context.Context, obj runtime.Object) error
   Create(ctx context.Context, obj runtime.Object) error
   Get(ctx context.Context, key ObjectKey, obj runtime.Object) error
   Delete(ctx context.Context, obj runtime.Object, opts ...DeleteOptionFunc) error
}
```

This will be beneficial:
- No need to refactor your code because the fake client can't handle your use case.
- No need to write unit test(s) for your internal refactored code.

If you look at `CatalogSourceConfigCRUD` it's very similar to the `Client` interface above.

Reply:
Isn't this what `client-gen` provides in some sense? Can't we just use that? Or won't it help in this case?

Reply:
`client-gen` also generates the client interfaces. `operator-sdk` doesn't do it by default though. But yeah, this is exactly what I am referring to.

Reply:
I suggest we go with `client-gen` then.

Reply:
@aravindhp I used `client-gen` to generate a typed clientset, and it looks like it might not help us in this case. 

For starters, the client used in the clientset is of type `k8s.io/client-go/rest.Interface` where as the client used by marketplace is of type `sigs.k8s.io/controller-runtime/pkg/client.Client`. If we were to use the typed clientset in the refactored code, we'd have to change the client that marketplace is currently using. I think @tkashem  was suggesting more of an interface that uses the current client instead of changing the client we use. 
Also, `client-gen` generates interfaces that are tied to `types`, while @tkashem's suggestion was a `generalized client wrapper`

Reply:
In that case it is fine not to use `client-gen`